lib/fluent/plugin/in_system_metrics.rb

```
@@ -0,0 +1,143 @@
+            path_key = path == '/' ? 'root' : path.gsub(/[^\w]/, '_')
+            metrics["disk.#{path_key}.total_bytes"] = total_bytes
+            metrics["disk.#{path_key}.free_bytes"] = free_bytes
+            metrics["disk.#{path_key}.available_bytes"] = available_bytes
+            metrics["disk.#{path_key}.used_bytes"] = used_bytes
+            metrics["disk.#{path_key}.usage_percent"] = 100.0 * used_bytes / total_bytes if total_bytes > 0
+          end
+        rescue => e
+          log.error "Failed to collect disk metrics", error: e
+        end
+        metrics
+      end
```

RuboCop (1.75.5)

[convention] Assignment Branch Condition size for collect_disk_metrics is too high. [<14, 23, 7> 27.82/23]

(Metrics/AbcSize)